### ISSUE TITLE
Drop external uuid dependency in favor of native functionality

### ISSUE THREAD

**Issue by vhscom** [sveltejs/kit#4929]
### Describe the problem

The default demo [currently uses](https://github.com/sveltejs/kit/blob/d723e876be53db66f0549def417d94a3c0014653/packages/create-svelte/templates/default/src/hooks.ts#L1) a dev dependency for generating random v4 UUIDs inside the hooks file which has been available natively in Node and the Web Crypto API for some time now. In Node `crypto.randomUUID()` it [has been available](https://nodejs.org/docs/latest-v14.x/api/crypto.html#crypto_crypto_randomuuid_options) since v14.17.0. The Web Cryptography API, which appears to support web workers as well, uses the [same method name](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID).

### Describe the proposed solution

Dropping this dependency now that [support for Node 14 has ended](https://github.com/sveltejs/kit/issues/4916) seems like a good time to cut the baggage.

### Alternatives considered

Leave the dependency in place. It's [slower than native methods](https://dev.to/galkin/crypto-randomuuid-vs-uuid-v4-47i5) but probably not noticeable in the demo.

### Importance

nice to have

**Comment by Conduitry** [comment]
What happens when the app is deployed with an adapter to an environment that doesn't have the Node runtime available?

**Comment by ghostdevv** [comment]
Also luke's uuid package isn't exactly slow:

- <https://github.com/lukeed/uuid#benchmarks>
- <https://github.com/lukeed/uuid#performance>

**Comment by jrmoynihan** [comment]
Nanoid is even smaller, if I recall correctly.  Not sure if it's any faster though.  Probably not in a meaningful way.

**Comment by davecaruso** [comment]
ive been looking at a few issues, and i think this change should definitely be done after #4934 is merged, since it seems to include the web `crypto` global, where the example can use the web crypto method.

**Comment by vhscom** [comment]
> What happens when the app is deployed with an adapter to an environment that doesn't have the Node runtime available?

Thanks. I tried a Pages build after I saw your comment and your intuition was right. The build fails with a reference error as `crypto` import can't be used in the worker as, in the worker context, it's a global. But we can't currently remove the import statement in the hooks file (despite valid syntax otherwise) as a Node context is used for the build.

Curious what avenue @davecaruso is exploring as I'd prefer not to rely on a dependency for anything security-related when native functionality is available (the performance angle mentioned in OP is just icing on the cake).

**Comment by davecaruso** [comment]
> Thanks. I tried a Pages build after I saw your comment and your intuition was right. The build fails with a reference error as `crypto` import can't be used in the worker as, in the worker context, it's a global. And we can't currently remove the import in the hooks file (despite valid syntax) as `crypto` isn't available in the global namespace during the kit build.

this is an issue i currently face on my own site, but i have this hack in my svelte config
```ts
import { webcrypto } from 'crypto';
global.crypto = webcrypto;
```
(this works because the config is run only in node environments, and during dev and build its all sharing that global scope)

we obviously shouldn't throw this in the demo, but it's fine as a temp fix for other projects.
